bin/install_gems

```diff
@@ -1,16 +1,23 @@
 #!/bin/bash -eu
 
+dump_ruby_env_info "before running $(basename "$0")"
+
 ARCHITECTURE=$(uname -m)
 RUBY_VERSION=$(cat .ruby-version)
 GEMFILE_HASH=$(hash_file Gemfile.lock)
 CACHEKEY="$BUILDKITE_PIPELINE_SLUG-$ARCHITECTURE-ruby$RUBY_VERSION-$GEMFILE_HASH"
 
 restore_cache "$CACHEKEY"
 
-# Ensure we're using the latest version of Bundler before running bundle install.
-#
-# See https://github.com/Automattic/bash-cache-buildkite-plugin/issues/16  for more details on why this is recommended.
-gem install bundler
+# Install the same Bundler version as the one in the Gemfile.lock.
+# This should prevent runtime issues where Ruby "gets confused" on which Bundler version to use.
+# See https://github.com/Automattic/bash-cache-buildkite-plugin/issues/16 for more details.
+GEMFILE_LOCK_BUNDLER_VERSION=$(sed -n -e '/BUNDLED WITH/{n;s/ *//p;}' Gemfile.lock)
+# The command will fail if the `--version` parameter is empty; we can omit checking for that ourselves.
+gem install bundler --version "$GEMFILE_LOCK_BUNDLER_VERSION"
```

Comment:
@crazytonyli 's work in 114-gh-Automattic/buildkite-ci should address the issue this change works around at the root, but I think it's still useful to respect the Bundler version set in the Gemfile.lock.